baremetal/metal3machine_manager_test.go

```diff
@@ -488,7 +488,19 @@ var _ = Describe("Metal3Machine manager", func() {
 				},
 			},
 		}
-		host2 := *newBareMetalHost("myhost", nil, bmh.StateNone, nil, false, "metadata", false)
+		host2 := bmh.BareMetalHost{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "host2",
+				Namespace: namespaceName,
+				Labels:    map[string]string{"key1": "value1"},
+			},
+			Spec: bmh.BareMetalHostSpec{},
+			Status: bmh.BareMetalHostStatus{
+				Provisioning: bmh.ProvisionStatus{
+					State: bmh.StateReady,
+				},
+			},
+		}
```

Comment:
I believe that is already [there](https://github.com/metal3-io/cluster-api-provider-metal3/blob/b2f6a413390adbdf7cbcb9b33479372d23dd8ad9/baremetal/metal3machine_manager_test.go#L4472-L4476), when you set clusterlabel to True, i.e:

`host2 := newBareMetalHost("host2", nil, bmov1alpha1.StateReady, nil, false, "metadata", true)`

Reply:
availableHost (host2) is the one without labels. We already have a host with a label (hostWithLabel).  So, availableHost should not have any labels. If we create it with newBareMetalHost() function, it will have a lebel.

Reply:
> availableHost (host2) is the one without labels. We already have a host with a label (hostWithLabel). So, availableHost should not have any labels. If we create it with newBareMetalHost() function, it will have a lebel.

that is fine, you can toggle (not) to have label set just setting last input of  newBareMetalHost(), so if you need to set it:

`host2 := newBareMetalHost("host2", nil, bmov1alpha1.StateReady, nil, false, "metadata", true)`

otherwise:

`host2 := newBareMetalHost("host2", nil, bmov1alpha1.StateReady, nil, false, "metadata", false)`

Reply:
done